### PR TITLE
fix: :bug: encoding issues with member handle

### PR DESCRIPTION
**Pull request by ignazio-bovo**
### Issue
Due to the failed `utf-8` decoding attempt, the orion processor attempted to create a second member with the incorrectly decoded handle '��'
This caused a violation on the unique constraint on membership handle.

### Solution
- remove the constraint on the handle field which will be decoded from utf-8 if possible 
- add a new field `handleRaw` with the raw bytes from the handle with a unique constraint

### CODE REVIEW

File: src/mappings/membership/index.ts [hunk 1]
```diff
@@ -68,6 +70,11 @@ export async function processMemberProfileUpdatedEvent({
   }
 }
 
+function updateMemberHandle(member: Membership, newHandle: Uint8Array) {
+  member.handleRaw = '0x' + Buffer.from(newHandle).toString('hex')
```
Comment (zeeshanakram3) [inline src/mappings/membership/index.ts:74]: you can use the `u8aToHex` method from `polkadot/util`. It converts `Uint8Array` to hex string out of the box

```ts
import { u8aToHex } from '@polkadot/util'
```

**Review by mnaamani — APPROVED** [review]

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]

File: src/mappings/utils.ts [hunk 2]
```diff
@@ -24,7 +25,9 @@ videoRelevanceManager.init(1000 * 60 * 60)
 export const JOYSTREAM_SS58_PREFIX = 126
 
 export function bytesToString(b: Uint8Array): string {
-  return Buffer.from(b).toString()
+  const buffer = Buffer.from(b) // assumes default utf-8 encoding
+  // Use the utf-8-validate library to check if the Buffer contains valid UTF-8 data
+  return isValidUTF8(buffer) ? buffer.toString('utf-8') : buffer.toString()
```
[the remark this answers is not in the document]
Comment (ignazio-bovo) [reply src/mappings/utils.ts:30]: I meant to output a hex string in case of invalid utf-8 😅

[the remark this answers is not in the document]
Comment (ignazio-bovo) [reply src/mappings/utils.ts:30]: I meant to output a hex string in case of invalid utf-8 😅

File: src/mappings/membership/index.ts [hunk 3]
```diff
@@ -68,6 +71,11 @@ export async function processMemberProfileUpdatedEvent({
   }
 }
 
+function updateMemberHandle(member: Membership, newHandle: Uint8Array) {
+  member.handleRaw = '0x' + u8aToHex(newHandle)
```
Comment (zeeshanakram3) [inline src/mappings/membership/index.ts:75]: The `u8aToHex` function itself prepends the `0x` before the hex string
```suggestion
  member.handleRaw = u8aToHex(newHandle)
```

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Looks good. Just one last thing, please also bump the package version in the `package.json` file

**Review by zeeshanakram3 — APPROVED** [review]
LGTM

**Pull request merged**